packages/vue-urql/src/useQuery.ts

```diff
@@ -151,13 +190,24 @@ export function useQuery<T = any, V = object>(
     },
   };
 
-  return {
-    ...result,
+  const response: UseQueryResponse<T, V> = {
+    ...state,
     then(onFulfilled, onRejected) {
-      isThenable = true;
-      return executeQuery()
-        .then(() => result)
-        .then(onFulfilled, onRejected);
+      let result$: Promise<UseQueryState<T, V>>;
+      if (fetching.value && source.value) {
+        result$ = pipe(
+          source.value,
```

Comment:
Disclaimer: This might be my lack of understanding for wonka speaking:

What happens if, while the request of the first source is still running, a side-effect i.e. changes variables and a new source & request is created?

* Would this promise never resolve?
* Would this promise still resolve to the original requests value?
* Would this promise somehow resolve to the second request's value, as we concat sources on line 134?

The third behavior is what I think *should* happen but I can't tell if it does, and the tests don't cover it yet.

Reply:
It's the second case right now, but potentially we can now try shoehorning the third case in.